### DISCUSSION TITLE
Why am I getting a type error when using `String.split`?

### DISCUSSION THREAD

**Discussion by kronaemmanuel** [prisma/prisma#14065] [Q&A] [answered]
I have this code which works perfectly:
```
  const policies = await prisma.policy.findMany({
    where: {
      ...,
      AND: {
        status: {
          in: ["ACTIVE", "PENDING"]
        }
      }
    },
    select: {
     ...
    }
  })
```
However, the following code which should give the same result (imo at least) gives me an error:
```
  const str = "ACTIVE,PENDING";
  const policies = await prisma.policy.findMany({
    where: {
      ...,
      AND: {
        status: {
          in: str.split(",")
        }
      }
    },
    select: {
     ...
    }
  })
```
The error is as follows:
```
src/index.ts(39,7): error TS2322: Type '{ status: { in: string[]; }; }' is not assignable to type 'Enumerable<PolicyWhereInput> | undefined'.
  Types of property 'status' are incompatible.
    Type '{ in: string[]; }' is not assignable to type 'EnumPolicyStatusFilter | PolicyStatus | undefined'.
      Types of property 'in' are incompatible.
        Type 'string[]' is not assignable to type 'Enumerable<PolicyStatus> | undefined'.
          Type 'string[]' is not assignable to type 'PolicyStatus[]'.
            Type 'string' is not assignable to type 'PolicyStatus'.
```

This might not exactly be an issue with Prisma exactly (probably due to Typescript?), but I would love to get some help on this.

**Comment by ruheni** [comment, marked as answer]
Hey 👋🏽

It seems you're using an enum for the `status` field. In the Prisma Client generation step, Prisma Client will resolve the enums in your Prisma schema as [TypeScript String enums](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) which is different from the string type. You can take a look under the hood to see the TypeScript types Prisma generates. 🙂 

Don't hesitate to let me know if you would like any further clarification.

**Reply by kronaemmanuel** [reply]
Hey, Thanks a lot for your suggestion. I looked into it and here's what I did to solve this issue:

1. Import the required enum type from Prisma: `import {PolicyStatus} from '@prisma/client'`
2. Replace my code with:
```
  const str = "ACTIVE,PENDING";
  const policies = await prisma.policy.findMany({
    where: {
      ...,
      AND: {
        status: {
          in: str.split(",") as PolicyStatus[]
        }
      }
    },
    select: {
     ...
    }
  })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
